scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,136 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survival_guide = dsp.survival_guide or {}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+-- 1 = Original zones
+-- 2 = Zilart Zones
+-- 4 = Chains of Promethia Zones
+-- 8 = Treasures of Aht Urgahn
+-- 16 = Wings of the Goddess
+-- 2048 = Seekers of Auldoin
+local function getEnabledExpansions()
+    -- Original areas and zilart are always available.
+    local returnValue = 1 + 2
+
+    if (ENABLE_COP == 1) then returnValue = returnValue + 4 end
+
+    if (ENABLE_TOAU == 1) then returnValue = returnValue + 8 end
+
+    if (ENABLE_WOTG == 1) then returnValue = returnValue + 16 end
+
+    if (ENABLE_SOA == 1) then returnValue = returnValue + 2048 end
+
+    return returnValue
+end
+
+local function checkForRegisteredSurvivalGuide(player, guide, masks)
+    local group = guide.group
+    local currentZoneId = player:getZoneID()
+    local mask = nil
+    local variableSuffix = ''
+    local hasRegisteredGuide = false
+
+    if (guide.groupIndex == 32) then
+        mask = masks[group + 4]
+        variableSuffix = 'a'
+
+        -- This turns negative in the database for int32 max
+        hasRegisteredGuide = mask == (guide.groupMask * -1)
+    else
+        mask = masks[group]
+        hasRegisteredGuide = (bit.band(mask, guide.groupMask) ~= 0)
+    end
+
+    if (not hasRegisteredGuide) then
+        player:PrintToPlayer(
+            'This guide hasn\'t been registered yet, registering this Survival Guide.',
+            13)
+
+        local updatedValue = mask + guide.groupMask
+
+        if (variableSuffix == 'a') then
+            updatedValue = guide.groupMask
+            masks[group + 4] = updatedValue
+        else
+            masks[group] = updatedValue
+        end
+
+        player:setVar("SgTeleportMask" .. group .. variableSuffix, updatedValue)
+    end
+
+    return masks
+end
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+dsp.survival_guide.onTrigger = function(player, csid, index)
+    local guide = survivalGuides[index]
+
+    if (not (guide == nil) or not (ENABLE_ROV == 1)) then
```

Comment:
Good points, being new to Lua I didn't know how that would evaulate.

I think I copied the wrong thing. Was looking at regimes for this type of idea. Since there isn't a setting specifically for Survival Guides and there is a stigma against new settings I will just make sure a guide was found.